File: 412.py
# LeetCode 412 Fizz Buzz
# Reference: https://www.youtube.com/watch?v=GfVoxKFpDV0

class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        
        answer = []

        for i in range(1, n+1):
            if i % 3 == 0 and i % 5 == 0:
                temp = "FizzBuzz"
            elif i % 3 == 0:
                temp = "Fizz"
            elif i % 5 == 0:
                temp = "Buzz"
            else:
                temp = str(i)
                
            answer.append(temp)
        
        return answer

File: 412.java
// LeetCode 412 Fizz Buzz
// Reference: https://leetcode.com/problems/fizz-buzz/solutions/2628136/java-3-approaches-easy/?envType=featured-list&envId=top-interview-questions?envType=featured-list&envId=top-interview-questions

import java.util.ArrayList;

class Solution {
    public List<String> fizzBuzz(int n) {

        // No % method
        List<String> res = new ArrayList<>();

        int i = 1, fizz = 0, buzz = 0;

        while(i <= n){
            fizz++;
            buzz++;

            if (fizz == 3 && buzz == 5){

                res.add("FizzBuzz");
                fizz = buzz = 0;

            } else if (fizz == 3){

                res.add("Fizz");
                fizz = 0;

            } else if (buzz == 5){

                res.add ("Buzz");
                buzz = 0;
            } else {

                res.add(String.valueOf(i));
            }
            
            i++;
        }
        
        return res;
    }
}